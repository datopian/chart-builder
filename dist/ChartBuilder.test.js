"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _react2 = require("@testing-library/react");

var _ChartBuilder = require("./ChartBuilder");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var view = {
  resources: [{
    name: 'test',
    schema: {
      fields: [{
        name: 'a',
        type: 'string'
      }, {
        name: 'b',
        type: 'string'
      }, {
        name: 'c',
        type: 'string'
      }]
    }
  }]
};
it('renders UI', function () {
  var _render = (0, _react2.render)(_react.default.createElement(_ChartBuilder.ChartBuilder, {
    view: view,
    dataViewBuilderAction: function dataViewBuilderAction() {}
  })),
      container = _render.container;

  expect(container).toMatchSnapshot();
});
it('shows error message for non-compiled view', function () {
  var view = {};

  var _render2 = (0, _react2.render)(_react.default.createElement(_ChartBuilder.ChartBuilder, {
    view: view,
    dataViewBuilderAction: function dataViewBuilderAction() {}
  })),
      container = _render2.container;

  expect(container).toMatchSnapshot();
});
it('shows error message if no schema', function () {
  var view = {
    resources: [{
      name: 'no-schema'
    }]
  };

  var _render3 = (0, _react2.render)(_react.default.createElement(_ChartBuilder.ChartBuilder, {
    view: view,
    dataViewBuilderAction: function dataViewBuilderAction() {}
  })),
      container = _render3.container;

  expect(container).toMatchSnapshot();
});