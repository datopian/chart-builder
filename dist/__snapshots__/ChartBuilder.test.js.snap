// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI 1`] = `
<div>
  <div
    class="text-center w-full max-w-lg"
  >
    <form
      class="bg-white"
    >
      <div
        class="flex flex-wrap -mx-3 mb-2"
      >
        <div
          class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="chartType"
          >
            Chart type
          </label>
          <div
            class="relative"
          >
            <select
              class="block appearance-none w-full bg-gray-200 border border-gray-200 text-gray-700 py-3 px-4 pr-8 rounded leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
              name="chartType"
              placeholder="Chart type"
              required=""
            >
              <option
                value="line"
              >
                line
              </option>
              <option
                value="bar"
              >
                bar
              </option>
            </select>
            <div
              class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
            >
              <svg
                class="fill-current h-4 w-4"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="xAxis"
          >
            Group column
          </label>
          <div
            class="relative"
          >
            <select
              class="block appearance-none w-full bg-gray-200 border border-gray-200 text-gray-700 py-3 px-4 pr-8 rounded leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
              name="xAxis"
              placeholder="Field for X axis"
              required=""
            >
              <option
                value="a"
              >
                a
              </option>
              <option
                value="b"
              >
                b
              </option>
              <option
                value="c"
              >
                c
              </option>
            </select>
            <div
              class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700"
            >
              <svg
                class="fill-current h-4 w-4"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-1/3 px-3 mb-6 md:mb-0"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="yAxis"
          >
            Series
          </label>
          <label
            class="block text-gray-500 font-bold"
          >
            <input
              checked=""
              class="mr-2 leading-tight"
              name="yAxis"
              type="checkbox"
              value="a"
            />
            <span
              class="text-sm"
            >
              a
            </span>
          </label>
          <label
            class="block text-gray-500 font-bold"
          >
            <input
              class="mr-2 leading-tight"
              name="yAxis"
              type="checkbox"
              value="b"
            />
            <span
              class="text-sm"
            >
              b
            </span>
          </label>
          <label
            class="block text-gray-500 font-bold"
          >
            <input
              class="mr-2 leading-tight"
              name="yAxis"
              type="checkbox"
              value="c"
            />
            <span
              class="text-sm"
            >
              c
            </span>
          </label>
        </div>
      </div>
      <div
        class="flex items-center"
      >
        <div
          class="w-1/3"
        />
        <div
          class="w-1/3"
        >
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
            type="submit"
          >
            Add view
          </button>
        </div>
        <div
          class="md:w-1/3"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`shows error message for non-compiled view 1`] = `
<div>
  <div>
    ChartBuilder requires resource to be compiled into view.
  </div>
</div>
`;

exports[`shows error message if no schema 1`] = `
<div>
  <div>
    ChartBuilder requires resource schema.
  </div>
</div>
`;
